src/NClient.Providers/NClient.Providers.Api.Rest/RestRequestBuilder.cs

```diff
@@ -77,12 +78,16 @@ internal class RestRequestBuilder : IRequestBuilder
                 .ToArray();
             var route = _routeProvider
                 .Build(routeTemplate, methodInvocation.Method.ClientName, methodInvocation.Method.Name, methodParameters, methodInvocation.Method.UseVersionAttribute);
+
+            var uri = (await host.TryGetUriAsync(cancellationToken).ConfigureAwait(false))?.ToString();
+            if (string.IsNullOrEmpty(uri))
+                throw _clientValidationExceptionFactory.HostUriNotDefined();
             
-            var resource = new Uri(PathHelper.Combine(host.ToString(), route));
+            var resource = new Uri(Path.Combine(uri, route));
```

Comment:
PathHelper is needed here because it prepares a resource to combine with a relative path

Reply:
Done.